### DISCUSSION TITLE
How to isolate trials from a specific variable for accuracy?

### DISCUSSION THREAD

**Discussion by ncfisherim** [jspsych/jsPsych#1173] [Q&A] [answered]
Hello,

I followed the tutorial on presenting the accuracy of responses to the participant, but the problem is that when I have multiple variables with trials, it counts all of them; therefore, instead of getting x/12 trials it calculates it to be something like 200/12.  How can I isolate it so that it only counts correct_trials from var irb_1practice  ?  Thank you in advance.

Here is the code for the debrief after completing the practice:

`  var practice_debrief = {
  type: "html-keyboard-response",
  stimulus: function() {

    var trials = jsPsych.data.get();
    var correct_trials = trials.filter({correct: true});
    var accuracy = Math.round(correct_trials.count() / 12 * 100);
    var rt = Math.round(correct_trials.select('rt').mean());
    return "<p>You responded correctly on "+accuracy+"% of the trials.</p>"+
    "<p>Feel free to take a break if needed.<br><br>Press space when you are ready to begin the actual task.</p>";
  },
  choices: [32],
};

`

**Comment by becky-gilbert** [comment, marked as answer]
Hi @ncfisherim! You can add more key-value pairs to jsPsych's [data `.filter` function](https://www.jspsych.org/core_library/jspsych-data/#filter) so that you're just getting back the trials you need. So for instance, if your `irb_1practice` trials are the only ones that use the image-keyboard-response plugin, then you could filter them on the `type` parameter, like this:
```javascript
var correct_trials = trials.filter({type: 'image-keyboard-response', correct: true});
```
Or if your practice trials aren't set up in a way that makes them easy to filter, you could add a flag to the data for these trials using the trial object's `data` parameter, e.g.
```javascript
var irb_1practice = {
  type: 'html-keyboard-response',
  data: {task_segment: 'irb_1practice'},
  // ... other trial parameters
}
```
And then you can filter these trials like this:
```javascript
var correct_trials = trials.filter({task_segment: 'irb_1practice', correct: true});
```
I hope that helps!

**Reply by ncfisherim** [reply]
This is exactly what I needed.  Thank you so much!
